### PR TITLE
Added catalog of green architectural tactics for ML-enabled systems

### PR DESCRIPTION
**Pull request by xJREB**
This PR adds the 30 green tactics for ML-enabled systems from our ICSE SEIS 2024 paper.

### CODE REVIEW

**Comment by xJREB** [comment]
@iffatfatima, could you maybe have a very brief look to check if I made any major mistakes (e.g., a file type in a wrong location)? I had to move most categories to the tags because it seems to be one category per tactic collection, but I'm fairly confident that I did it correctly. Thank you!

File: docs/_posts/green-ml-enabled-systems/2023-12-13-choose-an-energy-efficient-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+layout: tactic
+
+title:  "Choose an Energy Efficient Algorithm"
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-choose-an-energy-efficient-algorithm.md:4]: remove one extra space after title:

File: docs/_posts/green-ml-enabled-systems/2023-12-13-consider-knowledge-distillation.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+layout: tactic
+
+title:  "Consider Knowledge Distillation"
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-consider-knowledge-distillation.md:4]: remove one extra space after title:

File: docs/_posts/green-ml-enabled-systems/2023-12-13-apply-cloud-fog-network.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+t-type: "Architectural Tactic"
+categories: green-ml-enabled-systems
+t-description: "Devices at the edge are usually connected to distant cloud services. However, bringing the cloud closer to edge devices could be more energy efficient. A cloud fog network (CFN) is one way to achieve that. CFN supports an architecture where deep neural network models are processed in servers between end-devices and clouds. For example, Yosuf et al. (2021) present a CFN architecture that consists of four layers: IoT end devices, Access Fog (AF), Metro Fog (MF), and Cloud Datacenter (CDC). This architecture led to a 68% reduction in power consumption when compared to a traditional cloud data center architecture on average."
+t-participant: "Software Designer"
+t-artifact: "Machine Learning Algorithm"
+t-context: "Network"
+t-feature: 
+t-intent: "Improve energy efficiency by using CFNs to bring the cloud closer to edge devices and reduce energy consumption due to data transfers"
+t-targetQA: "Energy Efficiency"
+t-relatedQA:"Performance"
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-apply-cloud-fog-network.md]: add space after t-relatedQA:

File: docs/_posts/green-ml-enabled-systems/2023-12-13-choose-an-energy-efficient-algorithm.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+t-description: "Different machine learning algorithms have different levels of energy consumption and computational power. For example, the K-nearest neighbor (KNN) algorithm has a much higher energy consumption than Random Forest (RF) (Kaack et al., 2022). High energy consumption does not necessarily mean that those algorithms perform better or achieve higher accuracy levels than low-energy algorithms. "
+t-participant: "Data Scientist"
+t-artifact: "Algorithm"
+t-context: "Machine Learning"
+t-feature: "Inference"
+t-intent: "Improve energy efficiency by choosing an energy-efficient algorithm that can achieve wanted model outcomes"
+t-targetQA: "Energy Efficiency"
+t-relatedQA: "Accuracy"
+t-measuredimpact: "Choosing suitable, energy efficient algorithms that achieve wanted outcomes can reduce the energy consumption of ML models (Kaack et al., 2022)"
+t-source:
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-choose-an-energy-efficient-algorithm.md]: remove extra space after t-source:

File: docs/_posts/green-ml-enabled-systems/2023-12-13-remove-redundant-data.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+t-participant: "Data Scientist"
+t-artifact: "Data"
+t-context: "Machine Learning"
+t-feature: 
+t-intent: "Enhance energy efficiency by detecting and removing redundant data to reduce the size of input data"
+t-targetQA: "Energy Efficiency"
+t-relatedQA: "Accuracy, Data Representativeness"
+t-measuredimpact: "Removing redundant data from the dataset leads to a smaller input data that further decreases computation, computational time, energy consumption, and memory space"
+t-source: "Priyadarshan Dhabe, Param Mirani, Rahul Chugwani, and Sadanand Gandewar. 2021. Data Set Reduction to Improve Computing Efficiency and Energy Consumption in Healthcare Domain. In Digital Literacy and Socio-Cultural Acceptance of ICT in Developing Countries. Springer, 53–64; Phyllis Ang, Bhuwan Dhingra, and Lisa Wu Wills. 2022. Characterizing the Efficiency vs. Accuracy Trade-off for Long-Context NLP Models. In Proceedings of NLP Power! The First Workshop on Efficient Benchmarking in NLP. Association for Computational Linguistics, Dublin, Ireland, 113–121." https://aclanthology.org/2022.nlppower-1.12"
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-remove-redundant-data.md]: remove extra " before https://aclan... in the last line

File: docs/_posts/green-ml-enabled-systems/2023-12-13-consider-knowledge-distillation.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+tags: machine-learning model-optimization
+t-sort: "Awesome Tactic"
+t-type: "Architectural Tactic"
+categories: green-ml-enabled-systems
+t-description: "Knowledge distillation is a technique where a large, complex model (teacher) is used to train a smaller, simpler model (student). The goal is to transfer the learned information from the teacher model to the student model, allowing the student model to achieve comparable performance while requiring fewer computational resources. "Knowledge distillation improves performance when evaluating top 5 accuracy and energy consumption"
```
Comment (iffatfatima) [inline docs/_posts/green-ml-enabled-systems/2023-12-13-consider-knowledge-distillation.md]: remove the " before Knowledge distillation in the last sentence.

**Comment by xJREB** [comment]
Many thanks for the detailed review and finding these things! :) I fixed them all and will merge now.

**Review by iffatfatima — COMMENTED** [review]
The clickable URLs in t-source are not working atm. May need to change some things in the template.

**Pull request merged**